### PR TITLE
migrate OpenJDK 1.8 and 11 from travis to github actions

### PR DESCRIPTION
**Pull request by bradh**
Also includes QA checks

As requested on mailing list, starts the migration away from Travis CI.

## Checklist

For all pull requests:

- [X] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [X] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [X] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [X] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer (there is an automatic PR check verifying this, check this when it turns green).

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [N/A] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [X] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [X] PR for bug fixes and small new features are presented as a single commit
- [N/A] Commit message(s) must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [N/A] New unit tests have been added covering the changes
- [X] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [X] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [N/A] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    - name: Maven repository caching
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    - name: Build with Maven
+      run: |
+        mvn -B clean install -Dfmt.skip=true -Dall -T2 --file pom.xml
```
[the remark this answers is not in the document]
Comment (bradh) [reply .github/workflows/linux.yml:26]: Will fix. Just doing the geoserver part at the moment.  Looks like I have the pip part working on the docs for that, will copy it across.

[the remark this answers is not in the document]
Comment (bradh) [reply .github/workflows/linux.yml:26]: Also, This should probably be considered draft for a while - would like to clean it up a bit more.

**Comment by mprins** [comment]
I can take a stab at some of the remaining online cases (mssql/mysql/oracle). Not sure if oracle gonna  work due to the size of the docker image, but worth a shot.

File: modules/library/render/src/test/java/org/geotools/renderer/lite/LabelDisplacementModeTest.java [hunk 2]
```diff
@@ -462,12 +460,7 @@ public void testMultiLineLabelDisplacementY() throws Exception {
         String refPath =
                 "./src/test/resources/org/geotools/renderer/lite/test-data/displacementMode/testMultiLineLabelDisplacementY.png";
         // java 11 makes chars quite a bit bolder
-        int tolerance = 1300;
-        if (SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_11)
-                || SystemUtils.IS_OS_MAC_OSX
-                || SystemUtils.IS_OS_WINDOWS) {
-            tolerance = 2000;
-        }
+        int tolerance = 2000;
```
Comment (aaime) [inline modules/library/render/src/test/java/org/geotools/renderer/lite/LabelDisplacementModeTest.java:463]: What happened here, Java 8 is showing the same behavior?

**Comment by aaime** [comment]
Considering also @mprins  work, at this point the travis.yml file can be removed, right?
Besides the confilict on that file, this PR looks good to be merged (to me at least).

**Comment by mprins** [comment]
agree, travis build matrix should be empty after merging in or rebasing to master

File: modules/library/render/src/test/java/org/geotools/renderer/lite/LabelDisplacementModeTest.java (hunk 2 above)
Comment (bradh) [reply modules/library/render/src/test/java/org/geotools/renderer/lite/LabelDisplacementModeTest.java:463]: Yes. I saw ~1950 as the error.

My guess is that it isn't really a Java 11 rendering algorithm difference, but perhaps a font selection difference. So what you see depends on what font it gets (which in turn is a function of the font fallback options, and what fonts are installed). I'm optimistic that a follow-up activity can make this test more robust.

**Comment by bradh** [comment]
> Considering also @mprins work, at this point the travis.yml file can be removed, right?
> Besides the confilict on that file, this PR looks good to be merged (to me at least).

There is a trusty + oraclejdk8 test that I haven't migrated. I think trusty is too old (its out of support, but not actually EOL) to take forward, but I'm willing to listen to options. In particular, I'm considering not removing it in backport branches.

**Comment by bradh** [comment]
> agree, travis build matrix should be empty after merging in or rebasing to master

Removed `.travis.yml`.

File: modules/library/render/src/test/java/org/geotools/renderer/lite/LabelDisplacementModeTest.java (hunk 2 above)
Comment (aaime) [reply modules/library/render/src/test/java/org/geotools/renderer/lite/LabelDisplacementModeTest.java:463]: The tests are loading up Vera fonts at the beginning, and using them in the SLDs, so it should be using those exact fonts. But I agree it should be looked at in follow up.

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
+name: Linux with OpenJDK 1.8 and OpenJDK 11 GitHub CI
```
Comment (aaime) [inline .github/workflows/linux.yml:1]: Everything looks good, just have a hesitation about this title... it's long and makes it harder to find at a glance a particular build in the GitHub PR checks:

image

Just calling it "Linux" or "Linux GitHub CI" would probably be enough, the name of the job is added at the end and clarifies which bit of the build is executed. E.g. may look like:

``Linux GitHub CI / openjdk11 (pull request)"

**Review by mprins — CHANGES_REQUESTED** [review]
LGTM,
One thing I noticed is that the Travis Q&A build builds more modules than this PR builds
I'm missing the following modules:

```
[INFO] Web Processing Service ............................. SUCCESS [ 16.907 s]
[INFO] SOLR module ........................................ SUCCESS [ 26.967 s]
[INFO] Elasticsearch plugin ............................... SUCCESS [ 30.412 s]
[INFO] Geobuf DataStore ................................... SUCCESS [ 27.082 s]
[INFO] Flatgeobuf DataStore ............................... SUCCESS [ 15.544 s]
[INFO] Cartographic CSS parser ............................ SUCCESS [ 37.931 s]
[INFO] Next Generation JDBC DataStores .................... SUCCESS [  0.209 s]
[INFO] S3 GeoTIFF plugin .................................. SUCCESS [ 17.008 s]
[INFO] ArcSDE plugin ...................................... SUCCESS [  0.366 s]
[INFO] ArcSDE dummy api ................................... SUCCESS [ 12.885 s]
[INFO] ArcSDE support classes ............................. SUCCESS [ 22.939 s]
[INFO] ArcSDE DataStore plugin ............................ SUCCESS [ 56.448 s]
[INFO] GeoTools Documentation ............................. SUCCESS [01:08 min]
```

when comparing eg. https://travis-ci.org/github/geotools/geotools/jobs/746126050#L19937-L20069 with https://github.com/geotools/geotools/pull/3236/checks?check_run_id=1461146174#step:5:11848 so anything past  "gt-cql-json" which makes sense as travis runs:

 > `mvn -B -fae -Dall clean install -T2 -Dfmt.action=check -Dqa=true -DskipTests=true`

and GHA runs 

> `mvn -B clean install -Dfmt.action=check -Dqa=true -DskipTests=true --file pom.xml` 

So essentially the `-Dall` argument is missing in this PR's Q&A build

**Comment by bradh** [comment]
> So essentially the `-Dall` argument is missing in this PR's Q&A build

Yes. For unknown reasons I added it to the GeoServer build (since fixed) and missed it here.

Should now be resolved - I see those modules in the QA build.

**Review by mprins — APPROVED** [review]

**Comment by mprins** [comment]
if you want I you (or I) can cherry-pick this into #3241 and  #3242 which backport the relevant online tests to 23.x and 24.x

**Comment by bradh** [comment]
Squashed and merged.

@mprins I would appreciate if you can please cherry-pick or otherwise include in backports.

**Pull request merged**